tests/mlir_printer_test.py

```diff
@@ -0,0 +1,249 @@
+from io import StringIO
+from xdsl.ir import Attribute, Data, MLContext, MLIRType, Operation, ParametrizedAttribute
+from xdsl.irdl import (AnyAttr, ParameterDef, RegionDef, VarOperandDef,
+                       VarResultDef, irdl_attr_definition, irdl_op_definition)
+from xdsl.parser import Parser
+from xdsl.printer import Printer
+
+import re
+
```

Comment:
I think we should do differential testing here. We can do one flow that uses the converter to print and one flow that uses the new printer, then the python bindings parser and prints again. This way, even the names of the SSAValues should be the same and we don't have to think about what the result should be, but we could just feed any kind of program.

Reply:
Yeah that's a good idea! I'll add this once I'll manage to make the mlir bindings work on my machine ;)

Reply:
Do you have any concrete problems? I guess that either Michel or I already encountered most of the issues the bindings can cause ;S

Reply:
Oh, it's just that I haven't done this in a while, so I need to correctly configure MLIR again ;)
So no concrete problems at least!

Reply:
Actually, did you already got this error: `ImportError: cannot import name 'ir' from 'mlir._mlir_libs._mlir'` ?
I can `import mlir`, but I get an error when I `import mlir.ir`

Reply:
Are you sure it tries to import from the newest bindings? On my local machine, the import works fine. Any idea @webmiche ?

Reply:
Works on my machine. Are you on the commit in the README? Maybe the structure of the python bindings has changed...

Reply:
I would prefer to stay on that commit btw, since the python bindings in my other project have the same structure as the commit in the README.

Reply:
Yeah, when trying on another machine (with the right commit this time), it worked.
Also, it seems that it was probably because of a PYTHONPATH error, since this fixed it. Thanks!